crates/gitbutler-oxidize/src/ext.rs

```diff
@@ -0,0 +1,123 @@
+    ) -> Result<bool>;
+
+    /// Return default label names when merging trees.
+    ///
+    /// Note that these should probably rather be branch names, but that's for another day.
+    fn default_merge_labels(&self) -> gix::merge::blob::builtin_driver::text::Labels<'static> {
+        gix::merge::blob::builtin_driver::text::Labels {
+            ancestor: Some("base".into()),
+            current: Some("ours".into()),
+            other: Some("theirs".into()),
+        }
+    }
+
+    /// Tree merge options that enforce undecidable conflicts to be forcefully resolved
+    /// to favor ours, both when dealing with content merges and with tree merges.
+    fn merge_options_force_ours(&self) -> Result<gix::merge::tree::Options>;
+
+    /// Return options suitable for merging so that the merge stops immediately after the first conflict.
+    /// It also returns the conflict kind to use when checking for unresolved conflicts.
+    fn merge_options_fail_fast(
+        &self,
+    ) -> Result<(
+        gix::merge::tree::Options,
+        gix::merge::tree::TreatAsUnresolved,
+    )>;
+
+    /// Just like [`Self::merge_options_fail_fast()`], but additionally don't perform rename tracking.
+    /// This is useful if the merge result isn't going to be used, and we are only interested in knowing
+    /// if a merge would succeed.
+    fn merge_options_no_rewrites_fail_fast(
+        &self,
+    ) -> Result<(gix::merge::tree::Options, TreatAsUnresolved)>;
+}
+
+impl GixRepositoryExt for gix::Repository {
+    fn for_tree_diffing(mut self) -> anyhow::Result<Self> {
+        let bytes = self.compute_object_cache_size_for_tree_diffs(&***self.index_or_empty()?);
+        self.object_cache_size_if_unset(bytes);
+        Ok(self)
+    }
+
+    fn merges_cleanly_compat(
+        &self,
+        ancestor_tree: git2::Oid,
+        our_tree: git2::Oid,
+        their_tree: git2::Oid,
+    ) -> Result<bool> {
+        self.merges_cleanly(
+            git2_to_gix_object_id(ancestor_tree),
+            git2_to_gix_object_id(our_tree),
+            git2_to_gix_object_id(their_tree),
+        )
+    }
+
+    fn merges_cleanly(
+        &self,
+        ancestor_tree: gix::ObjectId,
+        our_tree: gix::ObjectId,
+        their_tree: gix::ObjectId,
+    ) -> Result<bool> {
+        let (options, conflict_kind) = self.merge_options_no_rewrites_fail_fast()?;
+        let merge_outcome = self
+            .merge_trees(
+                ancestor_tree,
+                our_tree,
+                their_tree,
+                Default::default(),
+                options,
+            )
+            .context("failed to merge trees")?;
+        Ok(!merge_outcome.has_unresolved_conflicts(conflict_kind))
+    }
+
+    fn merge_options_force_ours(&self) -> Result<Options> {
+        Ok(self
+            .tree_merge_options()?
+            .with_tree_favor(Some(gix::merge::tree::TreeFavor::Ours))
+            .with_file_favor(Some(gix::merge::tree::FileFavor::Ours)))
+    }
```

Comment:
:heart:

Reply:
Thanks, I do try to make APIs so that they can be stable - builders are a must for this for truly add-only types.